platform/hale-platform.target

```diff
@@ -48,8 +48,8 @@
 		<unit id="de.fhg.igd.equinox.test.feature.feature.group" version="1.2.0.202203220819"/>
 	</location>
 	<location includeAllPlatforms="false" includeConfigurePhase="false" includeMode="slicer" includeSource="true" type="InstallableUnit">
-		<repository location="https://gitlab.wetransform.to/hale/hale-build-support/raw/d6435ab10b4a5cab971d1d7839c95b590542b662/updatesites/platform"/>
-		<unit id="eu.esdihumboldt.hale.platform.feature.group" version="4.1.0.i20230505"/>
+		<repository location="https://gitlab.wetransform.to/hale/hale-build-support/raw/e9436fa1b3c40b216f01daf20c3f21085e43b235/updatesites/platform"/>
+		<unit id="eu.esdihumboldt.hale.platform.feature.group" version="5.0.0.i20231025"/>
```

Comment:
@emanuelaepure10 I mean this. My guess is you only updated the commit sha after the last change but not the version of the platform feature to reflect whatever version is present in that update site.

Reply:
I see. somehow I forgot to push. I just did it now. Let's see the results :-)

Reply:
Any idea how to solve this

```
Cannot resolve project dependencies:
  Software being installed: eu.esdihumboldt.hale.app.cli.commands 5.1.0.qualifier
  Missing requirement: eu.esdihumboldt.hale.common.align 5.1.0.qualifier requires 'osgi.bundle; com.vladsch.flexmark 0.64.0' but it could not be found
  Cannot satisfy dependency: eu.esdihumboldt.hale.app.cli.commands 5.1.0.qualifier depends on: osgi.bundle; eu.esdihumboldt.hale.common.align 2.9.5
```

Reply:
The update site does not contain this bundle. See for instance the content of the related [artifacts.jar](https://gitlab.wetransform.to/hale/hale-build-support/blob/e9436fa1b3c40b216f01daf20c3f21085e43b235/updatesites/platform/artifacts.jar) (the XML file lists all plugins in the update site).

I guess you

1. Did the platform generation either w/o custom Eclipse home or w/o removing the old eclipse artifact (see last sentence on [this comment](https://wetransform.atlassian.net/browse/ING-3236?focusedCommentId=77898))
2. Did not test the update site in Eclipse with hale studio or forgot to take measures that the new update site is used in Eclipse despite mostly no version changes on the bundles
3. After running the script to update hale-build-support ignored the deleted files and did not commit the deletions

Reply:
@stempler I have probably missed [this comment](https://wetransform.atlassian.net/browse/ING-3236?focusedCommentId=77898)

`In version 3 I changed the default eclipse artifact that is downloaded from the minimal eclipse product to a full Eclipse 2023-09. For this to be picked up you need to clear the folder .gradle/bnd-platform/ in your home directory.`

I somehow don't have a  .gradle/bnd-platform/ folder
![image](https://github.com/halestudio/hale/assets/67077116/defbea7d-2e04-4100-91f1-1dcf3967ef4b)
Why I don't have it? What should I do to have it?

Reply:
> For this to be picked up you need to clear the folder `.gradle/bnd-platform/` in your home directory.

In your **home directory** / **user home**.

In a *nix shell it would be the folder `~`,  if you use a Windows shell there seems to be a variable [%USERPROFILE%](https://en.wikipedia.org/wiki/%25USERPROFILE%25)

See https://en.wikipedia.org/wiki/Home_directory

Reply:
@stempler @kapil-agnihotri 
Please tell me what in this workflow is wrong or what is missing:
1. clean` .gradle/bnd-platform/`
2. move to `master ``https://github.com/halestudio/hale-platform `
  2.1. change `eclipseHome`
  2.1.0. I don't run  `./gradlew -p modules/resources`  because it was done by @stempler in a previous PR merged into the master and for me this fails
  2.2. run `./gradlew`
  2.3. run `  ./update-build-support.sh`
3. create PR for hale-platform-update with the changes in updatesite
4. create PR for hale-platform  with the version-history.json
5. in hale plugin platform 
  5.1. go to `hale-platform.target` update location repository with the commit from hale-platform-update.
  5.2. reload locations and reload target platform
  5.3. validate `HALE.product`
  5.4. commit the changes
 Thank you

Reply:
> change eclipseHome

This ideally should not be required, but as we found out with some Eclipse versions, including the one used by bnd-platform for some years, when running the update site creation there are bundles missing.
We changed the default artifact that is downloaded to one that works, but it does not automatically replace any previous downloaded artifact. Thus the cleanup in `~/.gradle/bnd-platform/`.

> run   ./update-build-support.sh

Ideally before doing that test the update site within Eclipse using a file URL to the built update site in the file system. Eclipse has some caching mechanisms that can be hindering in such cases, so it is recommended to first rename the update site folder to something unique (e.g. append date and a counter).

After testing and then updating the build-support repo via the script, all changes in build-support should be added to the commit. If there are unexpected changes, e.g. bundles removed but not replaced or very many changes even though only a few are expected, something could be going wrong and it makes sense to check/investigate.